tests/test_functions.py

```diff
@@ -0,0 +1,119 @@
+import requests, difflib, sys
+import re
+
+command_line_arguments = []
+
+number_of_errors = 0
+
+
+def set_command_line_arguments(arguments):
+    global command_line_arguments
+    command_line_arguments = arguments
+    if "-resetalltests" in command_line_arguments:
+        test_print("resetting all tests by saving every request for future comparisons. You should not have run this unless you are working on the test suite implementation itself and have verified that all tests passed before the reset.")
+
+def test_print(string):
+    sys.stdout.write("[synbiohub test] ")
+    sys.stdout.write(string)
+    sys.stdout.write("\n")
+
+# make html a little more human readable
+def format_html(htmlstring):
```

Comment:
Try BeautifulSoup for managing HTML (Formatting, parsing, etc.) https://www.crummy.com/software/BeautifulSoup/

Reply:
Good idea- I was actually looking at that and planning to do it in the future.
Travis would have to install the python package every time though if I understand correctly.

Reply:
Yep, that's fine. 
Right now, there is an `install` part of the testing stage which just contains a Docker build step. It could include a Python install step very easily.

You would just have to run `pip freeze > requirements.txt` inside your virtualenv and all the currently-installed packages would be documented. Then Travis just has to run `pip install -r requirements.txt`.